### ISSUE TITLE
`error:0A000417:SSL routines::ssl/tls alert illegal parameter` in version 3.2.1

### ISSUE THREAD

**Issue by J-roen** [bacularis/bacularis-docker#6]
Hello,

After upgrading `bacularis/bacularis-standalone` from `3.2.0-alpine` to `3.2.1-alpine`, I get this error in the Bacularis frontend:

![Image](https://github.com/bacularis/bacularis-docker/assets/21005771/84aeec3d-b180-4aeb-b88e-5eef93fb046a)

The response of `POST http://localhost:9097/web/monitor/` is:

```json
{
  "jobs": [],
  "running_jobs": [],
  "terminated_jobs": [],
  "pools": [],
  "clients": [],
  "jobtotals": [],
  "dbsize": [],
  "messages": [],
  "error": {
    "output": "Problem with connection to bconsole. Output=>Connecting to Director localhost:9101\nbconsole: message.c:1835-0 openssl.c:81 Connect failure: ERR=error:0A000417:SSL routines::ssl\/tls alert illegal parameter\nTLS negotiation failed\nDirector authorization problem.\nMost likely the passwords do not agree.\nIf you are using TLS, there may have been a certificate validation error during the TLS handshake.\nFor help, please see http:\/\/www.bacula.org\/rel-manual\/en\/problems\/Bacula_Frequently_Asked_Que.html, Exitcode=>1",
    "error": 4
  }
}
```

After downgrading to `3.2.0-alpine`, the frontend is working perfectly again.

Can you help me finding out what's going wrong? Thanks in advance!

J-roen

**Comment by ganiuszka** [comment]
Hello @J-roen,

Thank you for reporting this problem.

This error comes from bconsole and it is because bconsole is not able to connect to Director. There can be a couple of reasons of this error. For start I would propose to do upgrade once again and check passwords if they match.

To do it you need to log in to the container with putting in place ``BACULARIS_CONTAINER_ID`` the real container ID:

```
# docker exec -it BACULARIS_CONTAINER_ID /bin/bash
```

2) After you could run these two commands:

```
# bdirjson -rDirector | grep Password
```
```
# bbconsjson -rDirector | grep Password
```

and check if passwords match.

Maybe this update changes something in passwords. We will see.

Thank you in advance for this checking and for letting us know about the result.

Best regards,
Marcin Haba (gani)

**Comment by J-roen** [comment]
Hi @ganiuszka,

Thanks for your quick reply!

Before upgrading, both commands returned the same password. Then I upgraded and the output of the first command now is the same as before, but the output of the second commands is different. I do not recognise this password and don't have it in my configuration files.

Any ideas?

Thanks very much for your help,
J-roen

**Comment by ganiuszka** [comment]
Hello @J-roen 

Thanks for your test. This result means that passwords are different, so there is no chance to work without taking an action of adapting passwords.

In Bacularis ``3.2.1`` we updated the base container image with Alpine. Previous Alpine version was ``3.18.3`` and now we have ``3.20``. In the ``3.20`` there is a newer Bacula version than in ``3.18.3`` and this change could cause this problem with passwords.

I reviewed both previous and the currect packages with ``bconsole.conf`` and the default passwords are following:

### Alpine 3.18.3
bacula-13.0.2-r1.apk:
```
Director {
  Name = build-3-18-x86_64-dir
  DIRport = 9101
  address = localhost
  Password = "a0oh00wt5ViC/J7IeyKku5uo0YaQyQUtehW7PQMUMyw0"
}
```

### Alpine 3.20
bacula-13.0.3-r0.apk:
```
Director {
  Name = build-3-20-x86_64-dir
  DIRport = 9101
  address = localhost
  Password = "uT6oVPYqdZNEI30ooNu6MS8leqYu3oUdIOY7fB76nOFp"
}
```

I don't know your upgrade Bacularis container procedure, but I guess that you store outside container the Bacula configuration to make it persistent between upgrades, right? In this case I think it might be good to check if ``bconsole.conf`` file is stored to be persistent too. If not, then it could be the answer that after upgrade new bconsole config file with the new password has been used and it no longer match to Bacula Director password.

Do you think that it can be this case?

Best regards,
Marcin Haba (gani)

**Comment by J-roen** [comment]
Hi @ganiuszka,

You're totally right! I did see that only Alpine was updated, but I didn't realize that the password could have changed as well, as this hasn't been the case in earlier upgrades of Bacula(ris) in the last half year either. Also, I wasn't aware that I was using the default password for bconsole. Everything is working again now, so that's great.  I'm also glad that I now know how to debug this, and I hope this GitHub issue helps others with the same problem. Thank you very much for your help, it is very much appreciated!

Regards,
J-roen
